dropshot/examples/colliding-error-type-names.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024 Oxide Computer Companyy
+
+//! Example of an API using a user-defined error type to serialize error
+//! representations differently from [`dropshot::HttpError`].
+
+use dropshot::endpoint;
+use dropshot::ApiDescription;
+use dropshot::ConfigLogging;
+use dropshot::ConfigLoggingLevel;
+use dropshot::RequestContext;
+use dropshot::ServerBuilder;
+
+mod latex {
+    use super::*;
+
+    /// Errors returned by pdflatex.
+    ///
+    /// Good luck figuring out what these mean!
+    #[derive(
+        Clone, Debug, serde::Serialize, serde::Deserialize, schemars::JsonSchema,
+    )]
+    pub enum Error {
+        /// An hbox is overfull.
+        OverfullHbox {
+            /// The amount of badness in the overfull hbox.
+            badness: usize,
+        },
+        /// Like an overfull hbox, except the opposite of that.
+        UnderfullHbox {
+            /// This one also has badness.
+            badness: usize,
+        },
+    }
+
+    impl dropshot::error::IntoErrorResponse for Error {
+        fn into_error_response(
+            &self,
+            ctx: dropshot::error::ErrorContext<'_>,
+        ) -> http::Response<dropshot::Body> {
+            http::Response::builder()
+                .status(http::StatusCode::BAD_REQUEST)
+                .header("x-request-id", ctx.request_id)
```

Comment:
I see what you were saying -- yeah it really seems like we want Dropshot to do this rather than expect everybody will remember to do this stuff.